primitives/direct_rpc_server/src/lib.rs

```diff
@@ -0,0 +1,116 @@
+/*
+	Copyright 2019 Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(test, feature(assert_matches))]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+// re-export module to properly feature gate sgx and regular std environment
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+pub mod sgx_reexport_prelude {
+	pub use jsonrpc_core_sgx as jsonrpc_core;
+	pub use log_sgx as log;
+	pub use thiserror_sgx as thiserror;
+}
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+use crate::sgx_reexport_prelude::*;
+
+use crate::{rpc_connection_registry::ConnectionRegistry, rpc_watch_extractor::RpcWatchExtractor};
+use codec::{Decode, Error as CodecError};
+use serde_json::error::Error as SerdeJsonError;
+use std::vec::Vec;
+use substratee_worker_primitives::{RpcResponse, TrustedOperationStatus};
+use tls_websocket_server::{WebSocketConnection, WebSocketError};
+
+#[cfg(test)]
+mod mocks;
+
+#[cfg(test)]
+mod builders;
+
+mod rpc_connection_registry;
+pub mod rpc_responder;
+mod rpc_watch_extractor;
+pub mod rpc_ws_handler;
+
```

Comment:
With the two new crates, I tried to follow the design principle to have all public traits, errors ,etc in the `lib.rs` files, and to have as much modules private as possible.